### PR TITLE
fixing issue #456

### PR DESCRIPTION
**Pull request by yilongli**
...laceFirst/countOccurences

### CODE REVIEW

**Comment by yilongli** [comment]
@dwightguth please review

File: include/builtins/string.k [hunk 1]
```diff
@@ -118,16 +118,16 @@ module STRING-HOOKS
...
   rule #parseToken("#Int",S:String) => String2Int(S)
-  rule #parseToken("#Float",S:String) => (String2Float(S)) 
-  rule #parseToken("#String",S:String) => (S) 
+  rule #parseToken("#Float",S:String) => (String2Float(S))
+  rule #parseToken("#String",S:String) => (S)
   rule #parseToken("Int",S:String) => String2Int(S)
-  rule #parseToken("Float",S:String) => (String2Float(S)) 
-  rule #parseToken("String",S:String) => (S) 
+  rule #parseToken("Float",S:String) => (String2Float(S))
+  rule #parseToken("String",S:String) => (S)
```
Comment (andreistefanescu) [inline include/builtins/string.k:130]: What is the difference here?

**Comment by andreistefanescu** [comment]
lgtm

File: include/builtins/string.k (hunk 1 above)
Comment (yilongli) [inline include/builtins/string.k:130]: My editor removes trailing whitespaces automatically.

**Comment by yilongli** [comment]
@dwightguth could you make sure my implementation of `countAllOccurrences` is consistent with what is defined in K.

File: src/javasources/KTool/src/org/kframework/backend/java/builtins/BuiltinStringOperations.java [hunk 2]
```diff
@@ -153,4 +154,28 @@ public static StringToken directionality(StringToken term, TermContext context)
     public static StringToken token2string(UninterpretedToken token, TermContext context) {
         return StringToken.of(token.value());
     }
+    
+    public static StringToken replaceAll(StringToken source, StringToken regex,
+            StringToken replacement, TermContext context) {
```
Comment (dwightguth) [inline src/javasources/KTool/src/org/kframework/backend/java/builtins/BuiltinStringOperations.java]: You need to use StringUtils.replace, StringUtils.replaceOnce, etc.

The semantics of the K functions are as follows:

replace(input, pattern, replacement, n) replaces n times
replaceFirst replaces once
replaceAll replaces every instance

your countOcurrences looks correct though.

**Pull request merged**